components/server/src/projects/projects-service.ts

```diff
@@ -71,7 +78,8 @@ export class ProjectsService {
                 changeUrl: "changeUrl", // todo: compute in repositoryProvider
             });
         }
-        return result;
+        result.sort((a, b) => (b.changeDate || "").localeCompare(a.changeDate || ""));
+        return result.slice(0, 30);
```

Comment:
Question: Out of curiosity, why limit this to 30 in the backend, as opposed to in the front-end?

Reply:
Good point!